Mathlib/AlgebraicGeometry/Morphisms/Separated.lean

```diff
@@ -128,6 +127,40 @@ lemma IsSeparated.of_comp [IsSeparated (f ≫ g)] : IsSeparated f := by
 lemma IsSeparated.comp_iff [IsSeparated g] : IsSeparated (f ≫ g) ↔ IsSeparated f :=
   ⟨fun _ ↦ .of_comp f g, fun _ ↦ inferInstance⟩
 
+@[stacks 01KM]
+instance isClosedImmersion_equalizer_ι_left {S : Scheme} {X Y : Over S} [IsSeparated Y.hom]
+    (f g : X ⟶ Y) : IsClosedImmersion (equalizer.ι f g).left := by
+  refine IsClosedImmersion.stableUnderBaseChange
+    ((Limits.isPullback_equalizer_prod f g).map (Over.forget _)).flip ?_
+  rw [← MorphismProperty.cancel_right_of_respectsIso @IsClosedImmersion _
+    (Over.prodLeftIsoPullback Y Y).hom]
+  convert (inferInstanceAs (IsClosedImmersion (pullback.diagonal Y.hom)))
+  ext1 <;> simp [← Over.comp_left]
+
+/--
+Suppose `X` is a reduced scheme and that `f g : X ⟶ Y` agree over some separated `Y ⟶ Z`.
+Then `f = g` if `ι ≫ f = ι ≫ g` for some dominant `ι`.
+-/
+lemma ext_of_isDominant [IsReduced X] {f g : X ⟶ Y}
+    (s : Y ⟶ Z) [IsSeparated s] (h : f ≫ s = g ≫ s)
+    (ι : W ⟶ X) [IsDominant ι] (hU : ι ≫ f = ι ≫ g) : f = g := by
+  let X' : Over Z := Over.mk (f ≫ s)
+  let Y' : Over Z := Over.mk s
+  let U' : Over Z := Over.mk (ι ≫ f ≫ s)
+  let f' : X' ⟶ Y' := Over.homMk f
+  let g' : X' ⟶ Y' := Over.homMk g
+  let ι' : U' ⟶ X' := Over.homMk ι
```

Comment:
Only a side comment: If we do these translations more often, which I think is quite probable, it would be great to have an `overize` similar to `algebraize`.